payments-core/src/main/java/com/stripe/android/networking/StripeApiRepository.kt

```diff
@@ -1308,6 +1311,24 @@ internal class StripeApiRepository @JvmOverloads internal constructor(
         }
     }
 
+    override suspend fun paymentIntentLinkAccountSession(
+        paymentIntentLinkAccountSessionParams: PaymentIntentLinkAccountSessionParams,
+        requestOptions: ApiRequest.Options
+    ): LinkAccountSession? {
+        return fetchStripeModel(
+            apiRequestFactory.createPost(
+                url = getPaymentIntentLinkAccountSessionUrl(
+                    paymentIntentLinkAccountSessionParams.clientSecret
```

Comment:
It feels weird that we are passing a `clientSecret` when the url expects a `paymentIntentId`

Reply:
Is the `clientSecret` supposed to be passed in via post body perhaps?

Reply:
There's this nice function to extract the id from the secret actually - https://github.com/stripe/stripe-android/pull/4711/commits/f9a944980c5bfd7536fe58b5f76927a569c4c935

Reply:
We shouldn't rely on the format of the client secret, we should treat it as opaque. https://paper.dropbox.com/doc/Client-secret-format-on-mobile-SDKs-AsKA8cwgHGn456lQ4YptM

Reply:
Oh thanks for that, I wasn't aware. We'll be proposing a change in the `CollectBankAccountLauncher` contract to pass the entire PaymentIntent / SetupIntent instead of just the secret. iOS will be doing the same.

And in this PR, changing it to pass both id and secret if that makes sense: https://github.com/stripe/stripe-android/pull/4711/commits/0d57eb31318e5b361ca3e30da7179b1b8e44f076

Reply:
Let's discuss more on the design doc. I think requiring id and client secret makes sense, although it would be better to require only client secret. We usually don't receive the full object, it's created in the backend and there's no need to send everything to the client. Curious to see other people's opinions.

Reply:
Following up on https://paper.dropbox.com/doc/03-10-2022-collectBankAccount-Android-ACHv2-API-Review--Bd1XF94iTkSLD3nO2LmpwjkjAg-A57q21cKoCVw6XoJITL0Y?t=502941400348617556513829#:t=502941400348617556513829.